server/test/mutex.cpp

```diff
@@ -0,0 +1,159 @@
+#include <atomic>
+#include <catch2/catch.hpp>
+#include <lib/mutex.h>
+#include <thread>
+#include <vector>
+
+TEST_CASE("Lock/Unlock storage Mutex")
+{
+    bool                th1    = false;
+    std::atomic<size_t> result = 0;
+    std::vector<std::thread *>  pool;
+
+    pool.reserve(10);
```

Comment:
just note, reserve didn't clear memory... in release mode here could be crap, but you delete it later.